src/cmd/run.go

```diff
@@ -363,7 +363,13 @@ func runCommandWithFallbacks(container string, command []string, emitEscapeSeque
 				}
 			}
 		default:
-			return nil
+			var newerr error
+			if err != nil {
+				newerr = fmt.Errorf("failed to invoke command %s in container %s (Exit code: %d): %s", command[0], container, exitCode, err)
+			} else {
+				newerr = fmt.Errorf("failed to invoke command %s in container %s (Exit code: %d)", command[0], container, exitCode)
```

Comment:
Wouldn't it be better if `enter` and/or `run` forwarded the exit code of the requested command? eg., `toolbox run false` would have an exit code of 1 and `toolbox run true` would have 0.

I suppose, most people use the exit code programmatically because they can already see the standard error and output streams as a human. In that case, not forwarding the exit code breaks the programmability of it.

But it's hard to say without knowing the exact use-case.

Part of the complication in forwarding the exit code is that Cobra, the Go framework we use for our command line handling, only lets us return an error. See `Execute` in `src/cmd/root.go`. So, I think we will need to provide our own custom `error` implementation that can encapsulate an exit code.